### ISSUE TITLE
React Migration Plan

### ISSUE THREAD

**Issue by CowMuon** [hicommonwealth/commonwealth#2433]
~~Given that Jake is gone at the end of this week, we need to make sure we have a structured plan for completing the Interop layer while he is away, without anyone being blocked by any knowledge that may reside only in his head.~~
~~Acceptance Criteria: A structured plan for completion either in Notion, or as stories/tasks attached to this milestone.~~

Work plan for remaining items / stretch goals:

- [x] Merging master into React "still working" with minimum conflicts
- [x] Rewire component kit  (make modals avail where needed) - S5 not including modals/infinite scroll 
- [x] modals @alexyoung23j  has branch (login is its own subtask here) 
- [x] infinite scroll (?) 
- [x] poppers, menus* both looking good (no redirects yet) 
- [x] Finishing "last" of components - (using setState so now have exs of hooks using atoms)
- [x] Layout page rendering in general need attention (it breaks sometimes- why?) 
- [x] Use class component wrapper around poppers to localize state in order to minimize "functional creep" 
- [x] Poppers state cleanup (components using modals MUST be React components (proposed plan follows)
- [x] Use pure React instead of lodash for debouncing? 
- [x] Fix links across site `/navigation` (anyone could tackle this, but involves custom domains)
- [x] **Solve for custom domains**
- [ ] **Fix Quill rendering / Quill generally**
- [ ] **Ensure redraws work across site, esp when driven by controller layer (bad pattern generally -- could replace with event emitter)**
- [ ] **Ensure data layer does not break / explode (shouldn't, but we should keep an eye out for errors or improper request handling)**
- [ ] THEN we need the list of everything that's gone into master that needs to be updated on react branch (since we stopped merging master in.) 

_Proposed plan for poppers state cleanup_:  (convert atomic components to functional components before replacing popovers)
1. merge this incremental progress on popovers https://github.com/hicommonwealth/commonwealth/pull/2570
2. icons PR
3. text PR
4. buttons PR
5. inputs PR
6. tooltips PR
7. try popover replacement again

OR:
YA class component wrapper around functional poppers. (I'm not sure how that would work.) 

**Quill React options**:
https://github.com/zenoamaro/react-quill
https://www.npmjs.com/package/react-quilljs

**Modals** 
Get all little modals working then big Modal with everything it touches.

**Comment by CowMuon** [comment]
Is this correct and up-to-date? @zakhap @jnaviask @affordances

**Comment by affordances** [comment]
It's just kind of a heap of stuff with no real organization. Let's leave it for now and replace it all with the results of the Reactathon on Wednesday.

**Comment by CowMuon** [comment]
**Remaining areas of concern:** 
1. Remaining type errors- @bonnichiwa fixing profiles related & avatar upload; @affordances will catch rest (onCreate components, chain events, etc? )
2. Quill remaining render cases @raykyri?
3. Substrate removal- still some ambiguity about what's being removed? @jnaviask is confirming. 
4. No redraws in controller layer- @0xdantrinh fixed redraws in notifications controller, need to remove from remaining controllers (you can skip chat for now)
